### PR TITLE
Always clear the internal buffer after a call to Flush().

### PR DESCRIPTION
**Pull request by resospoons**
actually I don't really understand how this works in the "success" case, but this seems safe... I'm still digging around looking for a better solution

### CODE REVIEW

File: lib/go/thrift/http_client.go [hunk 1]
```diff
@@ -174,9 +174,23 @@ func (p *THttpClient) WriteString(s string) (n int, err error) {
...
+type bufferCloser struct {
+	*bytes.Buffer
+}
+
+// THttpClient reuses its buffer from one request to the next. If that
+// buffer is not reset, well, it'll just keep getting extended from
+// one request to the next. Make the bytes.Buffer into an io.Closer
+// and rely on the http transport to call Close() (and therefore
+// Reset()).
+func (b *bufferCloser) Close() error {
```
Comment (bhs) [inline lib/go/thrift/http_client.go:186]: nice and concise! thanks!

**Comment by bcronin** [comment]
For my own information:
- At one point our thrift repo was a proper subset of the apache thrift repo (as snapshotted at v0.9.2)
- It now is a divergent branch with fixes of our own

No implied anything in the above, just to want to confirm we're officially branching off the standard implementation at this point in case we ever upgrade versions or have the desire to make further changes / fixes.

**Comment by resospoons** [comment]
Yes. Changes are the 3 PRs here:
https://github.com/resonancelabs/thrift_go_0_9_2/commits/master

I made an Asana task to fold them back in.

**Comment by bcronin** [comment]
(I probably could have confirmed that myself had I been paying more attention to the pull request number being **3**. Oops.)

**Pull request merged**